### DISCUSSION TITLE
An update on mix-int solvers

### DISCUSSION THREAD

**Discussion by zoharl3** [yalmip/YALMIP#1010] [General discussion] [answered]
I'm currently looking a issue with gurobi:

https://support.gurobi.com/hc/en-us/community/posts/4407464930321-pre-solve-is-buggy?page=1#community_comment_4408516115601

I thought it we be nice to have an alternative. For example:

1. Mosek added support for mixed-int. Have you considered supporting it?

2. A while back, cplex stopped supporting matlab. Is this still the case?
Has no one written bindings? Probably not too difficult.

**Comment by johanlofberg** [comment, marked as answer]
Mosek has been fully supported since day 0

I don't follow developments on the cplex side. It is a sinking ship in my opinion.

**Reply by zoharl3** [reply]
Before I'll try to work out a minimal example, what could cause:

"Warning: Solver not applicable (mosek does not support integer variables)"
Problem -4: Solver not applicable

**Reply by johanlofberg** [reply]
(very) old version, or you are trying to solve an integer problem that Mosek does not support (i.e. MISDP)

**Reply by zoharl3** [reply]
I had a little mess in my problem formulation, which I can't begin to explain. Not sure how gurobi (or yalmip) accepted it up until now (and how it has worked so well, except for the occasional numerical issues).

For mixed-int, gurobi seems to perform significantly better than mosek.

About the sinking ship. Can you perhaps elaborate a bit? I recall that cplex used to perform on par with gurobi.

**Reply by johanlofberg** [reply]
Lots of people have left cplex development/management, which sort of smells like cplex isn't a priority within IBM. Dropping MATLAB support, takes ages to fix issues in their releases, horrible to sign up for academic license. Contrast to Gurobi and Mosek where the whole company exist only for the solver, rapid fixes and much more close to the user community.
